lib/src/material_color_picker.dart

```diff
@@ -77,47 +77,48 @@ class _MaterialColorPickerState extends State<MaterialColorPicker> {
     });
   }
 
-  ColorSwatch _findMainColor(Color shadeColor) {
-    for (final mainColor in _colors)
+  ColorSwatch? _findMainColor(Color shadeColor) {
+    for (final ColorSwatch? mainColor in _colors!)
       if (_isShadeOfMain(mainColor, shadeColor)) return mainColor;
 
-    return (shadeColor is ColorSwatch && _colors.contains(shadeColor))
+    return (shadeColor is ColorSwatch && _colors!.contains(shadeColor))
         ? shadeColor
         : null;
   }
 
-  bool _isShadeOfMain(ColorSwatch mainColor, Color shadeColor) {
-    for (final shade in _getMaterialColorShades(mainColor)) {
+  bool _isShadeOfMain(ColorSwatch? mainColor, Color? shadeColor) {
```

Comment:
Why is `mainColor` nullable here if we assert that it's not null in the first line of the function? In this case `mainColor` should not be nullable!